client/src/adapters/implementation/ChatDataAdapter.ts

```diff
@@ -1,17 +1,38 @@
 import { IChatDataAdapter } from '../interfaces/IChatDataAdapter.ts';
+import { MessageDataAdapter } from './MessageDataAdapter.ts';
+import MessageRepository from '../../api/MessageRepository.ts';
+import { Chat, ImageData, MessageData } from '../../types/types.ts';
+import ImageRepository from '../../api/ImageRepository.ts';
+import { ImageDataAdapter } from './ImageDataAdapter.ts';
 
 const ChatDataAdapter: IChatDataAdapter = {
-  getChats: (chatData) => {
-    return chatData.map((chat) => ({
-      id: chat._id,
-      chatName: '',
-      profilePicture: '',
-      lastMessage: '',
-      lastSender: '',
-      isLastMessageRead: false,
-      users: chat?.users || [],
-      messages: chat?.messages || [],
-    }));
+  getChats: async (chatData, currentUserId) => {
+    const chats: Chat[] = [];
+    for (const chat of chatData) {
+      const messagesResponse = await MessageRepository.getAllMessagesForChat(
```

Comment:
nie sprawdzamy czy response.data jest puste? bo wtedy ciężko cokolwiek zrobić

Reply:
chyba że się teraz mylę i to bangla na pustych danych, ale wtedy zwrotka się zesra na referencji po undefinedach